### ISSUE TITLE
SPWN_REQ error when running OSU Benchmarks 

### ISSUE THREAD

**Issue by dipti-kothari** [openpmix/prrte#996]
### What version of the PMIx Reference Server are you using? (e.g., v1.0, v2.1, git master @ hash, etc.)
Prrte: master @ 923745679d
##### What version of PMIx are you using? (e.g., v1.2.5, v2.0.3, v2.1.0, git branch name and hash, etc.)
Openpmix: master @ 10b03e44
### Please describe the system on which you are running

* Operating system/version: alinux2
* Computer hardware: AWS c5n.18xlarge instance
* Network type: EFA

-----------------------------

## Details of the problem

Hitting SPWN_REQ error when running osu benchmarks and using slurm 20.02.4 to generate hostfile. Outside of slurm I have not encountered this error
```
mpirun --prefix /fsx/ompi/install -n 576 --map-by ppr:36:node --output tag  --hostfile /fsx/hosts.file -x PATH -x LD_LIBRARY_PATH /fsx/osu-micro-benchmarks-5.6.3/libexec/osu-micro-benchmarks/mpi/collective/osu_allgather -x 10 -i 10
[ip-172-31-53-4:05839] PRTE ERROR: Not found in file prted/pmix/pmix_server_dyn.c at line 72
[ip-172-31-53-4:05839] UNABLE TO RETRIEVE SPWN_REQ FOR JOB mpirun-ip-172-31-53-4-5839@1 [room=4095]
[1,0]<stdout>:
[1,0]<stdout>:# OSU MPI Allgather Latency Test v5.6.3
[1,0]<stdout>:# Size       Avg Latency(us)
[1,0]<stdout>:1                     423.72
[1,0]<stdout>:2                     388.04
[1,0]<stdout>:4                     388.97
[1,0]<stdout>:8                     398.98
[1,0]<stdout>:16                    413.44
[1,0]<stdout>:32                    433.59
[1,0]<stdout>:64                    480.05
[1,0]<stdout>:128                   674.02
[1,0]<stdout>:256                   777.30
[1,0]<stdout>:512                  1234.88
[1,0]<stdout>:1024                 2227.44
[1,0]<stdout>:2048                 6747.69
[1,0]<stdout>:4096                 8827.58
[1,0]<stdout>:8192                10055.93
[1,0]<stdout>:16384               12532.17
[1,0]<stdout>:32768               17209.44
[1,0]<stdout>:65536               33756.21
[1,0]<stdout>:131072              57180.79
[1,0]<stdout>:262144             106500.93
[1,0]<stdout>:524288             207762.61
[ip-172-31-53-4:05839] PRTE ERROR: Not found in file prted/pmix/pmix_server_dyn.c at line 72
[ip-172-31-53-4:05839] UNABLE TO RETRIEVE SPWN_REQ FOR JOB mpirun-ip-172-31-53-4-5839@1 [room=4095]
```
This error started happening after prrte commit `8663bb49ea Improve node name matching`

**Comment by rhc54** [comment]
Pretty sure this has since been fixed - could you please update to current head of master and try again?

**Comment by dipti-kothari** [comment]
I tried it with the current head commits,  prrte master @ bf4955d7fc and openpmix master @ 10b03e44. I am still hitting the above error

**Comment by rhc54** [comment]
Okay, then I'll need some more info. From the prior problem, I assume `mpirun` is executing on one node while the hostfile only contains other nodes. Is this correct? Are you operating in a managed environment - i.e., are the nodes being allocated by a resource manager? Or are they solely defined by the hostfile?

Try adding `--prtemca prte_do_not_resolve 1` to your cmd line - does the message go away?

**Comment by dipti-kothari** [comment]
Yes, the error message does go away after adding `--prtemca prte_do_not_resolve 1` to mpirun. 

I am using AWS Parallel Cluster with the `initial_queue_size set` to 16(nodes are fixed) and using slurm as a scheduler. To generate the hostfile I am using slurm `srun -N x hostname > hostfile` and the contents of the hostfile is
``` 
compute-st-c5n18xlarge-2
compute-st-c5n18xlarge-5
compute-st-c5n18xlarge-3
compute-st-c5n18xlarge-4
compute-st-c5n18xlarge-1
```
I did verify, if the hostfile contains ip addresses there is no error.
I don't think I am answering your question regarding managed environment, as I am not entirely sure about it. Slurm is the resource manager here and Im using a hostfile as well.

**Comment by rhc54** [comment]
Just a question: is there some reason to use the hostfile in addition? We read the Slurm assignment, so I'm just curious why you specify the hostfile as well. If you remove the hostfile from your original cmd line, does the error message go away?

Also, since you specify `--prefix`, you don't need the `-x PATH -x LD_LIBRARY_PATH` options. The `prefix` option tells us to prepend that location to both `PATH` and `LD_LIBRARY_PATH` and Slurm automatically forwards your environment to the remote nodes. So the `-x` options really aren't doing anything here.

**Comment by rhc54** [comment]
Took me awhile to realize that while you have Slurm, you aren't actually using it except to create a hostfile. In other words, you aren't using it to manage anything. It's kinda odd, but I guess in your setup it doesn't much matter. However, it does make a difference in the code path here.

I'll take a gander at it. For now, just use that MCA param or the IP addresses.

**Comment by dipti-kothari** [comment]
Yes, that is correct! Thanks

**Comment by abouteiller** [comment]
I see a similar issue on Nersc Cori under managed Slurm, even with the `--prtemca prte_do_not_resolve 1` param. 

Cori is currently in maintenance; I am trying to replicate on another system where debugging is easier and will produce some stack trace/debug output if possible.

**Comment by hppritcha** [comment]
Not seeing issues in the cori MTT - at least since the last update of master.

**Comment by rhc54** [comment]
@dipti-kothari Could you please try again with new head of PMIx and PRRTE?

**Comment by wckzhang** [comment]
I'll take a look sometime this week

**Comment by rhc54** [comment]
Finally chased this down and fixed it in #1018
